src/libponyc/ast/parser.c

```diff
@@ -1052,11 +1052,8 @@ DEF(field);
   TOKEN("field name", TK_ID);
   SKIP("mandatory type declaration on field", TK_COLON);
   RULE("field type", type);
-  IF(TK_DELEGATE, RULE("delegated type", provides));
   IF(TK_ASSIGN, RULE("field value", infix));
-  // Order should be:
-  // id type value delegate_type
-  REORDER(0, 1, 3, 2);
+  AST_NODE(TK_NONE); // provides
```

Comment:
Removing the node and calls to it result in the `SugarTest.ObjectWithField` test failing with the following message:
```
Error:
Unexpected child node found, flet
    object let x: T = 3 fun foo() => 4 end
           ^
test/libponyc/util.cc:271: Failure
Value of: false
  Actual: false
Expected: true
test/libponyc/util.cc:317: Failure
Expected: check_ast_same(expect_package, package) doesn't generate new fatal failures in the current thread.
  Actual: it does.
test/libponyc/sugar.cc:1531: Failure
Expected: test_equiv(short_form, "sugar", full_form, "parse") doesn't generate new fatal failures in the current thread.
  Actual: it does.
```